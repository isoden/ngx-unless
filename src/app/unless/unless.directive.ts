import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';

@Directive({
  selector: '[ngxUnless]'
})
export class UnlessDirective {
  @Input()
  set ngxUnless(condition: any) {
    if (Boolean(condition)) {
      this.viewContainerRef.clear();
      return;
    }

    this.viewContainerRef.createEmbeddedView(this.templateRef);
  }

  constructor(
    private viewContainerRef: ViewContainerRef,
    private templateRef: TemplateRef<any>,
  ) { }

}
