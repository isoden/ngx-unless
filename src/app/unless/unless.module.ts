import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { UnlessDirective } from './unless.directive';

@NgModule({
  imports: [
    CommonModule,
  ],
  declarations: [
    UnlessDirective,
  ],
  exports: [
    UnlessDirective,
  ]
})
export class UnlessModule { }
