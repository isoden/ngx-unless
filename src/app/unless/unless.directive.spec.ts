import { Component } from '@angular/core';
import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { UnlessModule } from './unless.module';

@Component({ template: '' })
class TestComponent {
  condition = true;
}

function createTestComponent(template: string): ComponentFixture<TestComponent> {
  return TestBed.overrideComponent(TestComponent, {set: {template: template}})
      .createComponent(TestComponent);
}

describe('UnlessDirective', () => {
  let fixture: ComponentFixture<TestComponent>;

  function getComponent(): TestComponent { return fixture.componentInstance; }

  afterEach(() => fixture = null);

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [TestComponent],
      imports     : [UnlessModule],
    });
  });

  it('should work in a template attribute', async(() => {
    fixture = createTestComponent('<div *ngxUnless="condition">Hello1</div>');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  }));

  it('should work on a template element', async(() => {
    fixture = createTestComponent('<ng-template [ngxUnless]="condition">Hello2</ng-template>');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  }));

  it('should toggle node when condition changes', async(() => {
    fixture = createTestComponent('<div *ngIf="condition">hello3</div>');
    getComponent().condition = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    getComponent().condition = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    getComponent().condition = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  }));

});
