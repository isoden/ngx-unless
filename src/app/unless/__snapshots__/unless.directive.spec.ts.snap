// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnlessDirective should toggle node when condition changes 1`] = `
<ng-component
  condition="false"
>
  
</ng-component>
`;

exports[`UnlessDirective should toggle node when condition changes 2`] = `
<ng-component
  condition={[Function Boolean]}
>
  <div>
    hello3
  </div>
</ng-component>
`;

exports[`UnlessDirective should toggle node when condition changes 3`] = `
<ng-component
  condition="false"
>
  
</ng-component>
`;

exports[`UnlessDirective should work in a template attribute 1`] = `
<ng-component
  condition={[Function Boolean]}
>
  
</ng-component>
`;

exports[`UnlessDirective should work on a template element 1`] = `
<ng-component
  condition={[Function Boolean]}
>
  
</ng-component>
`;
